src/js/tools/drawing/Rectangle.js

```diff
@@ -16,11 +16,14 @@
 
   pskl.utils.inherit(ns.Rectangle, ns.ShapeTool);
 
-  ns.Rectangle.prototype.draw_ = function (col, row, color, targetFrame) {
+  /**
+   * @override
+   */
+  ns.Rectangle.prototype.draw = function (col, row, color, targetFrame) {
     var strokePoints = pskl.PixelUtils.getBoundRectanglePixels(this.startCol, this.startRow, col, row);
     for (var i = 0 ; i < strokePoints.length ; i++) {
       // Change model:
-      targetFrame.setPixel(strokePoints[i].col, strokePoints[i].row, color);
+      targetFrame.setPixel(strokePoints[i].col, strokePoints[i].row, this.getToolColor());
```

Comment:
I think you should use "color" instead of getToolColor().
The logic to swap TRANSPARENT_COLOR for the SELECTION_TRANSPARENT_COLOR is done in the ShapeTool here. It allows to preview your rectangle if you are using the transparent color.